api/xds_protocol.rst

```diff
@@ -469,3 +469,158 @@ occurred via a :ref:`resource update <xds_protocol_resource_update>`.
 
 .. |Multiple EDS requests on the same stream| image:: diagrams/eds-same-stream.svg
 .. |Multiple EDS requests on distinct streams| image:: diagrams/eds-distinct-stream.svg
+
+Implementations
+---------------
+
+.. _on_demand_vhds_rds_protocol:
+
+Virtual Host Discovery Service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+By default in RDS, all routes for a cluster are sent to every Envoy instance
+in the mesh. This causes scaling issues as the size of the cluster grows. The
+majority of this complexity can be found in the virtual host configurations, of
+which most are not needed by any individual proxy. 
+
+In order to fix this issue, the Virtual Host Discovery Service (VHDS) protocol
+uses the delta xDS protocol to allow a route configuration to be subscribed to
+and the necessary virtual hosts to be requested as needed. Instead of sending
+allmvirtual hosts with a route config, using VHDS will allow an Envoy instance
+to subscribe and unsubscribe from a list of virtual hosts stored internally in
+the xDS management server. The xDS management server will monitor this list and
+use it to filter the configuration sent to an individual Envoy instance to only
+contain the subscribed virtual hosts.
+
+In VHDS, Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with the 
+:ref:`type_url <envoy_api_field_DeltaDiscoveryRequest.type_url>` set to 
+`type.googleapis.com/envoy.api.v2.route.VirtualHost` 
+and :ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>` 
+set to a list of virtual host resource names for which it would like
+configuration. The management server will respond with a
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+with the :ref:`resources <envoy_api_field_DeltaDiscoveryResponse.resources>`
+field populated with the specified virtual hosts, the 
+:ref:`name <envoy_api_field_resource.name>` field populated with
+the virtual host name, and the 
+:ref:`aliases <envoy_api_field_resource.aliases>` field
+populated with the explicit (no wildcard) domains of the virtual host. Future
+updates to these virtual hosts will be sent via spontaneous updates.
+
+Updates to the route configuration entry to which a virtual host belongs will
+clear the virtual host table and require all virtual hosts to be sent again. It
+may be useful for the management server to populate and RDS responses with the
+subscribed list of virtual hosts. 
+
+Joining/Reconnecting to a management server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When an Envoy instance forms a delta xDS connection to the xDS management
+server for the first time, it will receive a base configuration filtered down
+to a subset of routes that are likely useful to the Envoy instance. For
+example, these might be filtered down by routes that exist in the current
+namespace. The xDS management server will send this same collection of
```

Comment:
What is a namespace? This is an Istio construct, can you provide a more generic term?

Reply:
I don't think Istio has any special meaning for namespace. I could say "these might be filtered down by routes that exist in the current Kubernetes namespace"?

Reply:
I could say "these might be filtered down by routes that exist in the current environment"

Reply:
I mean that it's a k8s construct; Envoy documentation should standalone, and not be dependent on any particular execution environment.

Reply:
Changed to: For example, these might be filtered down by routes that exist in the same application.